integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/database/DatabaseAwait.java

```diff
@@ -0,0 +1,72 @@
+package com.sequenceiq.it.cloudbreak.util.wait.service.database;
+
+import static com.sequenceiq.redbeams.api.model.common.Status.DELETE_COMPLETED;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.it.cloudbreak.RedbeamsClient;
+import com.sequenceiq.it.cloudbreak.context.RunningParameter;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.database.RedbeamsDatabaseServerTestDto;
+import com.sequenceiq.it.cloudbreak.log.Log;
+import com.sequenceiq.it.cloudbreak.util.ResponseUtil;
+import com.sequenceiq.it.cloudbreak.util.wait.service.Await;
+import com.sequenceiq.it.cloudbreak.util.wait.service.ExceptionChecker;
+import com.sequenceiq.it.cloudbreak.util.wait.service.Result;
+import com.sequenceiq.it.cloudbreak.util.wait.service.WaitResult;
+import com.sequenceiq.redbeams.api.model.common.Status;
+
+@Component
+public class DatabaseAwait extends Await<RedbeamsDatabaseServerTestDto, Status> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseAwait.class);
+
+    @Override
+    public RedbeamsDatabaseServerTestDto await(RedbeamsDatabaseServerTestDto entity, Status desiredStatus, TestContext testContext,
+            RunningParameter runningParameter, long pollingInterval, int maxRetry) {
+        try {
+            if (entity == null) {
+                throw new RuntimeException("Database key has been provided but no result in resource map!");
+            }
+            Log.await(LOGGER, String.format("%s for %s", entity.getResponse().getName(), desiredStatus));
+            RedbeamsClient client = testContext.getMicroserviceClient(RedbeamsClient.class, testContext.getWho(runningParameter)
+                    .getAccessKey());
+            String crn = entity.getResponse().getResourceCrn();
+            switch (desiredStatus) {
+                case AVAILABLE:
+                case STOPPED:
+                    waitForDatabaseStatus(new DatabaseOperationChecker<>(), client, crn, testContext, desiredStatus,
+                            pollingInterval, maxRetry);
+                    break;
+                case DELETE_COMPLETED:
+                    waitForDatabaseStatus(new DatabaseTerminationChecker<>(), client, crn, testContext, desiredStatus,
+                            pollingInterval, maxRetry);
+                    break;
+                default:
+                    LOGGER.warn("Wait checker is not implemented yet for the desired database state '{}' ", desiredStatus);
+                    break;
+            }
+            if (!desiredStatus.equals(DELETE_COMPLETED)) {
+                entity.refresh(testContext, null);
```

Comment:
Why are we refreshing with null?

Reply:
Here is the `CloudbreakClient` should be passed.
```awaitEntity.refresh(getTestContext(), (CloudbreakClient) null);}```
However that is not defined in `RedbeamsDatabaseServerTestDto await`. So we can remove this from `Redbeams Await`, because of Redbeams is
> "not refreshable resource" indeed